### PR TITLE
Proto version refactoring.

### PR DESCRIPTION
**Pull request by nickeskov**

### CODE REVIEW

File: cmd/retransmitter/retransmit/network/incoming_peer.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func RunIncomingPeer(ctx context.Context, params IncomingPeerParams) {
 	writeHandshake := proto.Handshake{
 		AppName: params.WavesNetwork,
 		// pass the same minor version as received
-		Version:      proto.Version{Major: 0, Minor: readHandshake.Version.Minor, Patch: 0},
```
Comment (alexeykiselev) [inline cmd/retransmitter/retransmit/network/incoming_peer.go:69]: Looks very strange. Why we always pass 0 as major version? I think we should only drop patch part from the incoming version.

File: cmd/binclient/main.go [hunk 2]
```diff
@@ -55,7 +52,7 @@ func main() {
 		zap.S().Fatal("please, provide 'version' CLI argument")
 	}
 
-	version, err := parseVersion(*version)
```
Comment (alexeykiselev) [inline cmd/binclient/main.go:58]: Those variables overwrites looks terrible

**Review by alexeykiselev — CHANGES_REQUESTED** [review]

File: cmd/retransmitter/retransmit/network/incoming_peer.go (hunk 1 above)
Comment (nickeskov) [reply cmd/retransmitter/retransmit/network/incoming_peer.go:69]: I agree. Fixed

File: cmd/binclient/main.go (hunk 2 above)
Comment (nickeskov) [reply cmd/binclient/main.go:58]: Also fixed

**Review by alexeykiselev — APPROVED** [review]

**Pull request merged**